hermes-management/src/main/java/pl/allegro/tech/hermes/management/config/storage/StorageConfiguration.java

```diff
@@ -82,18 +86,23 @@ MultiDatacenterRepositoryCommandExecutor multiDcRepositoryCommandExecutor(
     }
 
     @Bean
-    public SharedCounter sharedCounter() {
-        CuratorFramework curatorFramework = clientManager().getLocalClient().getCuratorFramework();
-        return new SharedCounter(curatorFramework,
+    SummedSharedCounter summedSharedCounter(ZookeeperClientManager manager) {
+        List<CuratorFramework> curatorClients = manager.getClients().stream()
+                .map(ZookeeperClient::getCuratorFramework)
+                .collect(toList());
+        return new SummedSharedCounter(
+                curatorClients,
                 storageClustersProperties.getSharedCountersExpiration(),
                 storageClustersProperties.getRetrySleep(),
                 storageClustersProperties.getRetryTimes());
     }
 
     @Bean
-    public DistributedEphemeralCounter distributedCounter() {
-        CuratorFramework curatorFramework = clientManager().getLocalClient().getCuratorFramework();
-        return new DistributedEphemeralCounter(curatorFramework);
+    SummedDistributedEphemeralCounter summedDistributedEphemeralCounter(ZookeeperClientManager manager) {
+        List<CuratorFramework> curatorClients = manager.getClients().stream()
```

Comment:
you could extract this (and from method above) to a private method like `getCuratorClients(manager)`

Reply:
Done